File: src/main/java/org/dxworks/githubminer/dto/response/invitations/Invitation.java
package org.dxworks.githubminer.dto.response.invitations;

import com.google.api.client.json.GenericJson;
import com.google.api.client.util.Key;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class Invitation extends GenericJson {
    @Key
    private BigDecimal id;
    @Key
    private GenericJson repository;
    @Key
    private String permissions;

    public String getRepoUrl() {
        return (String) getRepository().get("html_url");
    }
}

File: src/main/kotlin/org/dxworks/githubminer/dto/response/invitations/Invitation.kt
package org.dxworks.githubminer.dto.response.invitations

import com.google.api.client.json.GenericJson
import com.google.api.client.util.Key
import java.math.BigDecimal

class Invitation : GenericJson() {
    @Key
    lateinit var id: BigDecimal

    @Key
    var repository: GenericJson? = null

    @Key
    var permissions: String? = null

    val repoUrl: String?
        get() = repository?.get("html_url") as String?
}
